publisher.py

```diff
@@ -362,12 +363,16 @@ def fetch_image(self, image: Image):
         Each `Registry` has its own client so we just access it directly.
         """
         self.log.info(f'Fetching image {image}')
-        for line in image.repository.registry.client.pull(str(image.repository), image.tag, stream=True, decode=True):
+        imagerepo = str(image.repository)
+        imagerepo = imagerepo.replace('artifactory/api/docker/', '')
```

Comment:
yes, if has artifactory then only it will replace

Reply:
I am not comfortable for putting specific URLs in publisher. tomorrow we add support for newer registry, then we will be adding this kind URL again ?
See if there is a better way. else we can merge it.

Reply:
Dheeraj, we are passing  ( svl-artifactory.juniper.net/artifactory/api/docker/contrail-nightly-tmp ) URL and need to replace artifactory URI while passing to dockerclinet to pull. So it will replace this uri. It wont effect any other registry.